hal/src/sercom/v2/uart/mod.rs

```diff
@@ -0,0 +1,1005 @@
+    }
+
+    /// Update the UART [`Config`]uration.
+    ///
+    /// Calling this method will temporarily disable the SERCOM peripheral, as
+    /// some registers are enable-protected. This may interrupt any ongoing
+    /// transactions.
+    ///
+    /// ```
+    /// use atsamd_hal::sercom::v2::uart::{BaudMode, Oversampling, Uart};
+    /// uart.reconfigure(|c| c.set_run_in_standby(false));
+    /// ```
+    #[inline]
+    pub fn reconfigure<F>(&mut self, update: F)
+    where
+        F: FnOnce(&mut SpecificConfig<C>),
+    {
+        self._reconfigure(update);
+    }
+
+    /// Join [`RxDuplex`] and [`TxDuplex`] halves back into a full `Uart<C,
+    /// Duplex>`
+    pub fn join(rx: Uart<C, RxDuplex>, _tx: Uart<C, TxDuplex>) -> Self {
+        Self {
+            config: rx.config,
+            capability: PhantomData,
+        }
+    }
+}
+
+impl<C: ValidConfig> AsMut<Uart<C, Duplex>> for (&mut Uart<C, RxDuplex>, &mut Uart<C, TxDuplex>) {
+    #[inline]
+    fn as_mut(&mut self) -> &mut Uart<C, Duplex> {
+        // SAFETY: Pointer casting &mut Uart<C, RxDuplex> into &mut
+        // Uart<C, Duplex> should be safe as long as RxDuplex and Duplex
+        // both only have one nonzero-sized field.
+        unsafe { &mut *(self.0 as *mut _ as *mut Uart<C, Duplex>) }
+    }
+}
+
+impl<C, D> AsRef<SpecificConfig<C>> for Uart<C, D>
+where
+    C: ValidConfig,
+    D: Capability,
+{
+    #[inline]
+    fn as_ref(&self) -> &SpecificConfig<C> {
+        self.config.as_ref()
+    }
+}
+
+//=============================================================================
+// Rx/Tx specific functionality
+//=============================================================================
+
+impl<C, D> Uart<C, D>
+where
+    C: ValidConfig,
+    D: Receive,
+    DataSize: AsPrimitive<C::Word>,
+{
+    /// Read from the DATA register
+    ///
+    /// # Safety
+    ///
+    /// Reading from the data register directly is `unsafe`, because it will
+    /// clear the RXC flag, which could break assumptions made elsewhere in
+    /// this module.
+    #[inline]
+    pub unsafe fn read_data(&mut self) -> DataSize {
+        self.config.as_mut().registers.read_data()
+    }
+
+    /// Read the status register and convert into a [`Result`]
+    /// containing the corresponding [`Flags`] or [`Error`]
+    #[inline]
+    fn read_flags_errors(&self) -> Result<Flags, Error> {
+        self.read_status().try_into()?;
+        Ok(self.read_flags())
+    }
+
+    /// Flush the RX buffer and clear RX errors
+    #[inline]
+    pub fn flush_rx_buffer(&mut self) {
+        // TODO
+        // The datasheet states that disabling the receiver (RXEN) clears
+        // the RX buffer, and clears the BUFOVF, PERR and FERR bits.
+        // However, in practice, it seems like BUFOVF errors still pop
+        // up after a disable/enable cycle of the receiver, then immediately begin
+        // reading bytes from the DATA register.
+        //
+        // Is this a hardware bug???
```

Comment:
I think you should elevate this from a comment to part of the documentation. It seems important enough to tell users about.